src/MLJNearestNeighborsInterface.jl

```diff
@@ -1,36 +1,129 @@
 module MLJNearestNeighborsInterface
 
+# ===================================================================
+# IMPORTS
 import MLJModelInterface
-import MLJModelInterface: MODEL_TRAITS
+import MLJModelInterface: @mlj_model, metadata_model, metadata_pkg,
+    Table, Continuous, Count, Finite, OrderedFactor, Multiclass
+import NearestNeighbors
+import StatsBase
+import Tables
 
-using ScientificTypes
-using MLJBase
-import MLJBase: @load
-import MLJBase: Table, Continuous, Count, Finite, OrderedFactor, Multiclass
+using Distances
+using FillArrays
+using LinearAlgebra
+using Statistics
 
-using Requires, Pkg, Pkg.TOML, OrderedCollections, Parameters
-using Tables, CategoricalArrays, StatsBase, Statistics
-import Distributions
+# ===================================================================
+## EXPORTS
+export list_kernels, ColumnTable, DictTable
 
+# Export KNN models
+export KNNClassifier, KNNRegressor, KnnKernel, MultitargetKNNRegressor
 
+# Re-Export Distance Metrics from `Distances.jl`
+export Euclidean, Cityblock, Minkowski, Chebyshev, Hamming, WeightedEuclidean,
+    WeightedCityblock, WeightedMinkowski
 
+# Export KNN Kernels
+export DualU, DualD, Dudani, Fibonacci, Inverse, ISquared, KNNKernel, Macleod, Rank, ReciprocalRank,
+    UDK, Uniform , Zavreal
+     
 
-const srcdir = dirname(@__FILE__) # the directory containing this file
+# ===================================================================
+## CONSTANTS
+const Vec{T} = AbstractVector{T}
+const Mat{T} = AbstractMatrix{T}
+const Arr{T, N} = AbstractArray{T, N}
+const ColumnTable =  Tables.ColumnTable
+const DictTable = Dict{Symbol, <:AbstractVector}
```

Comment:
I guess we really should get Tables.jl to define this (cf ColumnTable) but I can live with this for now.